47.md

```diff
@@ -0,0 +1,112 @@
+
+## Rationale
+
+Paying zaps should be possible without the user needing to open a different app to only pay a Lightning invoice.
+This NIP describes a way for users to control a remote Lightning node or a custodial Lightning wallet. When self-hosting, this setup does not require the user to run their own server, thereby bypassing certain hurdles that are commonly encountered when trying to connect to a Lightning node remotely.
+
+## Terms
+
+* **client**: Nostr app on any platform that wants to pay Lightning invoices
+* **wallet service**: Nostr app that typically runs on an always-on computer (eg. in the cloud or on a Raspberry Pi).
+
+## Events
+
+There are two event kinds:
+- `NIP-47 request`: 23194
+- `NIP-47 response`: 23195
+
+Both the request and response events SHOULD only contain one `p` tag, containing the public key of the **wallet service** if this is a request, and the public key of the **client** if this is a response.
+
+The content is encrypted with [NIP04](https://github.com/nostr-protocol/nips/blob/master/04.md), and is a JSON object. The content depends on the kind.
+
+Request:
+```jsonc
+{
+    "cmd": "pay_invoice", // command, string
+    "data": { // data, object
+        "invoice": "lnbc50n1..." // command-related data
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "status": "ok", // status, "ok" | "error"
+    "event": "0123456789abcdef...", // event the command is in response to, string
+    "data": { // response data
+        "preimage": "0123456789abcdef..." // command-related data
+    }
+}
+```
+
+The data field SHOULD contain a `message` field with a human readable error message if the status is `error`.
+
+## Nostr Wallet Connect URI
+**client** discovers **wallet service** by scanning a QR code, handling a deeplink or pasting in a URI.
+
+The **wallet service** generates this connection URI with protocol `nostr+walletconnect:` and base path it's hex-encoded `pubkey` with the following query string parameters: 
+
+- `relay` Required. URL of the relay where the **wallet service** is connected and will be listening for events. May be more than one.
+- `secret` Required. 32-byte randomly generated hex encoded string. The **client** should use this to sign events when communicating with the **wallet service**.
+    - Authorization does not require passing keys back and forth.
+    - The user can have different keys for different applications. Keys can be revoked and created at will and have arbitrary constraints (eg. budgets).
+    - The key is harder to leak since it is not shown to the user and backed up.
+    - It improves privacy because the user's main key would not be linked to their payments.
+
+The **client** should then store this connection and use it when the user wants to perform actions like paying an invoice. Optionally it can display metadata about the connected **wallet service** from it's profile (name, image, url).
+
```

Comment:
Since this NIP uses ephemeral events, the relay used by the wallet service cannot terminate connections due to a lack of use from either the wallet service or the client side. Once a connection is established it should stay alive until the client closes it.